### ISSUE TITLE
Investigate task execution strategies?

### ISSUE THREAD

**Issue by bschroeter** [CABLE-LSM/benchcab#157]
Generic processing interface?
Graph dependency?

Mostly to run the parallel jobs, with task interdependence.

**Comment by SeanBryan51** [comment]
The main idea is we don't want to write the parallelisation code using `multiprocessing` more than once. For example:

https://github.com/CABLE-LSM/benchcab/blob/b5fe7fd12c2670c15a4d4b8269cfdccce521fede/benchcab/fluxsite.py#L375-L391

https://github.com/CABLE-LSM/benchcab/blob/b5fe7fd12c2670c15a4d4b8269cfdccce521fede/benchcab/comparison.py#L61-L81

We should define a general `Task` interface that has an abstract `run` method which gets implemented for fluxsite and comparison tasks.

**Comment by bschroeter** [comment]
This is largely speculative and exploratory. Voting heavy to account for the time involved.

**Comment by ccarouge** [comment]
I don't think we should do this now. Eventually, we will need to set something up within cylc so that it works with JULES. I would investigate this then.

**Comment by ccarouge** [comment]
Isn't this a duplicate of #156?

**Comment by SeanBryan51** [comment]
@ccarouge I would say this issue is related to https://github.com/CABLE-LSM/benchcab/issues/156 but not a duplicate. This issue is about the broader design approach for managing workflows in benchcab where as the other issue is addressing duplicate code when executing tasks in parallel.
